app/src/main/java/com/raduitache/myanimelist/seasonal/impl/SeasonalNavGraphImpl.kt

```diff
@@ -37,7 +43,20 @@ class SeasonalNavGraphImpl @Inject constructor(
 
     override fun NavGraphBuilder.buildNestedNavGraph(navController: NavController) {
         composable(startDestination.route, startDestination.namedNavArgs) {
-            startDestination.Content()
+            startDestination.Content {
+                navController.navigate("details/$it")
+            }
         }
+        composable(
+            "details/{details}",
+            arguments = listOf(navArgument("details") {type = NavType.StringType})
+        ) { backStackEntry ->
+            val animeDetailsViewModel = hiltViewModel<AnimeDetailsViewModel>()
+            backStackEntry.arguments?.getString("details")?.let { animeDetailsViewModel.selectAnime(it) }
+            AnimeDetailsScreen(animeDetailsViewModel) {
+                navController.popBackStack()
+            }
+        }
```

Comment:
I'd have this in a different navGraph, which is not a main nav graph. The bottom bar navigation shouldn't be visible when in an anime details view. Feel free to keep it as is for now, though.